### PR TITLE
Add support for Claude-3.5 Haiku

### PR DESCRIPTION
**Pull request by spolu**
## Description

Adds support for claude 3.5 haiku + remove claude 3  haiku

## Risk

None, tested in dev

## Deploy Plan

- deploy `front`
- run `npx tsx scripts/update_assistants_models.ts --fromModel claude-3-haiku-20240307 --toModel claude-3-5-haiku-20241022 --execute`

The product works fine during the migration interim (conversation can be seen and assitsants can be called on claude haiku even the assitsant builder shows it correctly, it's just gone from the picker)

### CODE REVIEW

File: types/src/front/lib/assistant.ts [hunk 1]
```diff
@@ -378,14 +381,28 @@ export const CLAUDE_3_5_SONNET_DEFAULT_MODEL_CONFIG: ModelConfigurationType = {
...
+};
 export const CLAUDE_3_HAIKU_DEFAULT_MODEL_CONFIG: ModelConfigurationType = {
   providerId: "anthropic",
   modelId: CLAUDE_3_HAIKU_20240307_MODEL_ID,
   displayName: "Claude 3 Haiku",
   contextSize: 180_000,
   recommendedTopK: 32,
   recommendedExhaustiveTopK: 128, // 65_536
-  largeModel: true,
+  largeModel: false,
```
Comment (spolu) [inline types/src/front/lib/assistant.ts:405]: I think this was inaccurate

File: types/src/front/lib/assistant.ts [hunk 2]
```diff
@@ -50,7 +50,7 @@ export function getSmallWhitelistedModel(
     return GPT_4O_MINI_MODEL_CONFIG;
   }
   if (isProviderWhitelisted(owner, "anthropic")) {
-    return CLAUDE_3_5_SONNET_DEFAULT_MODEL_CONFIG;
+    return CLAUDE_3_5_HAIKU_DEFAULT_MODEL_CONFIG;
```
Comment (spolu) [inline types/src/front/lib/assistant.ts:53]: I think this was inaccurate

Comment (fontanierh) [reply types/src/front/lib/assistant.ts:53]: It used to be the old sonnet here, which was then updated to new sonnet.

File: types/src/front/lib/assistant.ts [hunk 3]
```diff
@@ -378,14 +381,28 @@ export const CLAUDE_3_5_SONNET_DEFAULT_MODEL_CONFIG: ModelConfigurationType = {
...
   toolUseMetaPrompt: ANTHROPIC_TOOL_USE_META_PROMPT,
 };
+export const CLAUDE_3_5_HAIKU_DEFAULT_MODEL_CONFIG: ModelConfigurationType = {
+  providerId: "anthropic",
+  modelId: CLAUDE_3_5_HAIKU_20241022_MODEL_ID,
+  displayName: "Claude 3.5 Haiku",
+  contextSize: 180_000,
+  recommendedTopK: 32,
+  recommendedExhaustiveTopK: 128, // 65_536
+  largeModel: false,
```
Comment (fontanierh) [inline types/src/front/lib/assistant.ts:391]: note: it's 4x the price of the old haiku (1/3 the price of sonnet) so not that cheap

**Review by fontanierh — CHANGES_REQUESTED** [review]
LGTM but supportsVision needs to be false

File: types/src/front/lib/assistant.ts (hunk 3 above)
Comment (spolu) [reply types/src/front/lib/assistant.ts:391]: Yeah but we don't really rely on that that much. Only for legacy TEST_FREE_PLAN users which we plan to remove soon anyway :+1:

File: types/src/front/lib/assistant.ts [hunk 4]
```diff
@@ -378,14 +381,28 @@ export const CLAUDE_3_5_SONNET_DEFAULT_MODEL_CONFIG: ModelConfigurationType = {
...
+  description:
+    "Anthropic's Claude 3.5 Haiku model, cost effective and high throughput (200k context).",
+  shortDescription: "Anthropic's cost-effective model.",
+  isLegacy: false,
+  supportsVision: true,
```
[the remark this answers is not in the document]
Comment (spolu) [reply types/src/front/lib/assistant.ts:396]: I presume was also false for claude-3-haiku?

[the remark this answers is not in the document]
Comment (spolu) [reply types/src/front/lib/assistant.ts:396]: actually no claude-3-haiku do support vision

[the remark this answers is not in the document]
Comment (spolu) [reply types/src/front/lib/assistant.ts]: Thanks for catching!

**Review by fontanierh — APPROVED** [review]
LGTM

**Comment by spolu** [comment]
deployed and migration run

**Pull request merged**